src/sentry/api/endpoints/organization_events_timeseries.py

```diff
@@ -0,0 +1,371 @@
+    dataset: str
+    start: float
+    end: float
+
+
+class Row(TypedDict):
+    timestamp: float
+    value: float
+
+
+class SeriesMeta(TypedDict):
+    order: NotRequired[int]
+    groupBy: NotRequired[list[str]]
+    isOther: NotRequired[str]
+    unit: NotRequired[str]
+    type: str
+    interval: float
+
+
+class TimeSeries(TypedDict):
+    values: list[Row]
+    axis: str
+    meta: SeriesMeta
+
+
+class StatsResponse(TypedDict):
+    meta: StatsMeta
+    timeseries: list[TimeSeries]
+
+
+@region_silo_endpoint
+class OrganizationEventsTimeseriesEndpoint(OrganizationEventsV2EndpointBase):
+    publish_status = {
+        "GET": ApiPublishStatus.EXPERIMENTAL,
+    }
+
+    def get_features(
+        self, organization: Organization, request: Request
+    ) -> Mapping[str, bool | None]:
+        feature_names = [
+            "organizations:performance-use-metrics",
+            "organizations:dashboards-mep",
+            "organizations:mep-rollout-flag",
+        ]
+        batch_features = features.batch_has(
+            feature_names,
+            organization=organization,
+            actor=request.user,
+        )
+        return (
+            batch_features.get(f"organization:{organization.id}", {})
+            if batch_features is not None
+            else {
+                feature_name: features.has(
+                    feature_name, organization=organization, actor=request.user
+                )
+                for feature_name in feature_names
+            }
+        )
+
+    def get_request_querysource(self, request: Request, referrer: str) -> QuerySource:
+        if referrer in SENTRY_BACKEND_REFERRERS:
+            return QuerySource.SENTRY_BACKEND
+        else:
+            return self.get_request_source(request)
+
+    def get_top_events(self, request: Request) -> int:
+        if "topEvents" in request.GET:
+            try:
+                top_events = int(request.GET.get("topEvents", 0))
+            except ValueError:
+                raise ParseError(detail="topEvents must be an integer")
+            if top_events > MAX_TOP_EVENTS:
+                raise ParseError(detail=f"Can only get up to {MAX_TOP_EVENTS} top events")
+            elif top_events <= 0:
+                raise ParseError(detail="topEvents needs to be at least 1")
+
+            return top_events
+        else:
+            return 0
+
+    def get_comparison_delta(self, request: Request) -> timedelta | None:
+        if "comparisonDelta" in request.GET:
+            try:
+                return timedelta(seconds=int(request.GET["comparisonDelta"]))
+            except ValueError:
+                raise ParseError(detail="comparisonDelta must be an integer")
+        else:
+            return None
+
+    def get(self, request: Request, organization: Organization) -> Response:
+        with sentry_sdk.start_span(op="discover.endpoint", name="filter_params") as span:
+            span.set_data("organization", organization)
```

Comment:
i think our instrumentation shoud automatically cover a top level span and setting the organization  https://github.com/getsentry/sentry/blob/e7e1c643ea9d4a96edff8087bf9a91bf337cc493/src/sentry/api/bases/organization.py#L621

can probably delete this